dvc/stage/cache.py

```diff
@@ -166,7 +166,7 @@ def save(self, stage):
         dump_yaml(tmp, cache)
         self.tree.move(PathInfo(tmp), path)
 
-    def restore(self, stage, run_cache=True, pull=False):
+    def restore(self, stage, run_cache=True, pull=False, **kwargs):
```

Comment:
You could `pop` in `use_shell=kwargs.get("use_shell", True),` above and you won't have to make this change :slightly_smiling_face: